src/plug.ts

```diff
@@ -82,12 +97,41 @@ export class GlobalPlug implements Plug {
             return;
         }
 
+        const detectedAppId = detectAppId();
+        const configuredAppId = configuration.appId ?? null;
+
+        if (detectedAppId !== null && configuredAppId !== null && detectedAppId !== configuredAppId) {
+            throw new Error(
+                'The specified app ID and the auto-detected app ID are conflicting. '
+                + 'There is no need to specify an app ID when using an application-specific tag. '
+                + 'Please try again omitting the "appId" option.',
+            );
+        }
+
+        const appId = detectedAppId ?? configuredAppId;
+
+        if (appId === null) {
+            throw new Error(
+                'The app ID must be specified when it cannot be auto-detected. '
+                + 'Please try again specifying the "appId" option.',
+            );
+        }
+
         const {plugins, ...sdkConfiguration} = configuration;
-        const sdk = Sdk.init(sdkConfiguration);
+
+        const sdk = Sdk.init({
+            ...sdkConfiguration,
+            appId: appId,
+        });
 
         this.instance = sdk;
 
         const logger = this.instance.getLogger();
+
+        if (detectedAppId === configuredAppId) {
+            logger.warn('The "appId" option can be safely omitted when using the application-specific tag.');
```

Comment:
I think we should say that it is _preferred_ to be omitted in this case. Would help to propagate it as best practice. It also helps to copy-n-paste the plug initialization to other sites in the same organization without bothering with the appId, as it would already be loaded on the tag.

Reply:
Agreed. Any suggestions here?

Reply:
> Avoid specifying the "appId" option when using the application-specific tag as it can be detected automatically.

Reply:
I think it sound more of a command... Not sure... Maybe:
> Specifying the "appId" option should be avoided when using application-specific tag.

Reply:
> It is strongly recommended omitting the "appId" option when using the application-specific tag as it is can be detected automatically.

Reply:
> It is strongly recommended omitting the "appId" option when using the application-specific tag as it is automatically detected.

Reply:
> It is astronomically preferred and vastly recommended omitting the "appId" option when using the application-specific tag as it is detected automatically.